packages/flutter_tools/lib/src/globals.dart

```diff
@@ -2,6 +2,261 @@
 // Use of this source code is governed by a BSD-style license that can be
```

Comment:
I don't think we want this file changed since there's a conflicting and almost identical `globals_null_migrated`, so there will be duplicated symbols.  Instead, let's add `PreRunValidator get preRunValidator => context.get<PreRunValidator>() ?? const NoOpPreRunValidator();` to the bottom of `globals_null_migrated`.

Reply:
Here:
https://github.com/flutter/flutter/blob/3dab5550407a2c2fe70a762acfdfde93311148de/packages/flutter_tools/lib/src/globals_null_migrated.dart#L235